bots/qinnBot.py

```diff
@@ -0,0 +1,220 @@
+		CHARS = '!did not assign!'
+		words_panel = soup.find(id='wordPaneContent').get_text()
+		words_panel = words_panel.replace(u'\xa0', u' ')
+		# print('\n'+words_panel)
+		input_list = words_panel.rsplit(']')
+		DEFINITION = input_list[-1].strip()
+		del input_list[-1]
+		for sliced_string in input_list[::-1]:
+			# because of the below line(s) this function will break if the input chars parse into 19+ words
+			CHARS = sliced_string[sliced_string.rfind('0' or '1')+2:sliced_string.rfind('[')].strip()
+			PINYIN = sliced_string[sliced_string.rfind('[') + 2:None].strip()
+			ITERATOR = sliced_string[sliced_string.rfind('0' or '1'):sliced_string.rfind('0' or '1')+2].strip()
+			outputList[CHARS] = {'Simplified Form': CHARS, 'Pinyin': PINYIN, 'Definition': DEFINITION, 'Iterator': ITERATOR}
+			DEFINITION = sliced_string[1:(sliced_string.rfind('0' or '1') if (sliced_string.rfind('0' or '1')) > 0 else None)].strip()
+			# print('\n'+str(sliced_string))		
+		# print('\n'+str(outputList))
+		return outputList
+
+
+# discord-callable function to look up a bunch of words and parse them
+# needs some keyword arguments or something to include or disinclude info about a word
+# use wrappers for this? learn how to use wrappers properly? remember the word "exclude"?
+	@commands.command(name='qinnLookup', aliases=['QinnLookup','QinLookup','ql','qinlookup','权利'], 
+	help='Looks up Chinese words using the Archchinese online dictionary')
+	async def qinnLookup(self, ctx, char):
+		output_list = QinnBot.itemizeWordsButBetter(char)
+		output_string = str(output_list)
+		embed = discord.Embed(colour=discord.Colour.blurple()) #, description=output_string)
+		for key in output_list.keys():
+			embed.insert_field_at(0,
+				name='{0} : {1}'.format(key, output_list.get(key, 'eRrOr').get('Definition', 'bottom level error')), 
+				value='{1}'.format(output_list.get(key, 'top level error').get('Iterator', 'bottom level error'), output_list.get(key, 'top level error').get('Pinyin', 'bottom level error')),
+				inline=False)
+		await ctx.send(embed=embed)
+
+# discord-callable function to look up single character
+# is currently broken
+	@commands.command(name='qinnChar', aliases=['qinnchar','QinnChar','QinChar','qc','qinchar','清楚'], 
+	help='Looks up a single Chinese character using the Archchinese online dictionary')
+	async def qinnChar(self, ctx, chars):
+		print(chars)
+		for single_char in chars:
+			output_list = QinnBot.lookup(single_char)
+			output_string = str(output_list)
+			print(output_list)
+			embed = discord.Embed(colour=discord.Colour.blurple()) #, description=output_string)
+			embed.add_field(name='{} : {}'.format(single_char[:1], output_list.get('Definition', 'not found')),
+					value='{0}  |  Subchars: {2} {3}  |  Part of: {4}\n Encoding: {1}'.format(output_list.get('Pinyin'), output_list.get('Character Encoding'), str(output_list.get('Radical')).strip('. '), str(output_list.get('Component')).strip('. '), str(output_list.get('Part of')).strip('. ')),
```

Comment:
The .get function can take an optional argument that returns a value if the key is not found in the dictionary. Might be helpful to use that down here instead of using the !did not assign! thing earlier

Reply:
eh, I needed the '!did not assign!' for that specific function earlier, I could remove taht now, plus the default I want in the display funct is None anyhow, good to know though for sure